buffer/src/test/java/io/netty/buffer/AdaptiveByteBufAllocatorTest.java

```diff
@@ -158,4 +163,52 @@ public void run() {
             fail("Expected no exception, but got", throwable);
         }
     }
+
+    @SuppressWarnings("Since15")
+    @Test
+    @EnabledForJreRange(min = JRE.JAVA_17) // RecordingStream
+    @Timeout(10)
+    public void jfrChunkAllocation() {
+        try (RecordingStream stream = new RecordingStream()) {
+            CompletableFuture<RecordedEvent> future = new CompletableFuture<>();
+
+            stream.enable(AdaptivePoolingAllocator.AllocateChunkEvent.class);
+            stream.onEvent(AdaptivePoolingAllocator.AllocateChunkEvent.class.getName(), future::complete);
+            stream.startAsync();
+
+            AdaptiveByteBufAllocator alloc = new AdaptiveByteBufAllocator(true, false);
+            alloc.directBuffer(128).release();
```

Comment:
I think the `startAsync` is racing with this allocation, which would make the test flaky.

We could allocate in a loop with yield or sleep, until the future completes.

Reply:
Despite the name, startAsync does not actually start the recording asynchronously, it starts immediately. It's the event processing that's async

Reply:
Ah, ok. Makes sense.